### ISSUE TITLE
Feature request for horizontally placed legends

### ISSUE THREAD

**Issue by jmtilli** [timmolter/XChart#199]
I have developed a software that needs charting support, and I have considered XChart instead of JFreeChart due to not having to ship XChart source with the application (LGPL problems) and smaller size of XChart. However, there's one missing feature I would like to have.

Currently, if you do two line plots in the same chart, and display the legend using the position LegendPosition.OutsideS (outside below the chart), the legends are placed vertically. This means that lots of vertical space is wasted. This is completely unacceptable in my application where vertical space is limited.

I would like to have a feature to place the two legends horizontally, because the chart is quite wide and there is plenty of horizontal space below the chart. I place the legends outside and below the chart to eliminate the risk that they would be in the same area where useful data is showed.

**Comment by timmolter** [comment]
I've add the `feature request` tag. If you'd like to contribute the new feature via a pull request, I'd happily accept it. Otherwise I can try to do it if.when I have time. In the mean time, you could try placing the legend inside the chart if that works.

**Comment by jmtilli** [comment]
As an interim solution, I made the legend a .png file that is drawn into a custom JPanel placed with BorderLayout below the actual graph for which legend is disabled. So, this is no longer a show-stopper for me, but would be nice to have. I looked at the code, and it seems it would require so much knowledge of XChart internals that I'm not willing to spend the additional time considering that I found an interim solution.
